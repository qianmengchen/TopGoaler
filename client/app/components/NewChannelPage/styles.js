import { StyleSheet } from 'react-native';

const board = StyleSheet.create({
  container: {
    backgroundColor: 'grey',
    flex: 1,
    alignItems: 'stretch'
  },
  personalInfo: {
    flex: 0.35
  },
  channels: {
    flex: 0.2,
    backgroundColor: 'white'
  },
  tasks: {
    flex: 0.45,
    alignItems: 'stretch'
  }
});

export { board };
