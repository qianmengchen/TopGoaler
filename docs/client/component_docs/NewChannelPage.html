<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: NewChannelPage</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: NewChannelPage</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>NewChannelPage<span class="signature">()</span><span class="type-signature"></span></h2>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    

    
    <h4 class="name" id="NewChannelPage"><span class="type-signature"></span>new NewChannelPage<span class="signature">()</span><span class="type-signature"></span></h4>
    

    















    <h5 class="subsection-title">Properties:</h5>

    

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>userId</code></td>
            

            <td class="type">
            
                
<span class="param-type">number</span>


            
            </td>

            

            

            <td class="description last">A unique number identifying the user.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>addChannel</code></td>
            

            <td class="type">
            
                
<span class="param-type">function</span>


            
            </td>

            

            

            <td class="description last">A function allowing the user to create a new channel as the creator.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>navigation</code></td>
            

            <td class="type">
            
                
<span class="param-type">Object</span>


            
            </td>

            

            

            <td class="description last">An object that contains the navigate function allowing the user to navigate to other components.</td>
        </tr>

    
    </tbody>
</table>




<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="NewChannelPage_NewChannelPage.js.html">NewChannelPage/NewChannelPage.js</a>, <a href="NewChannelPage_NewChannelPage.js.html#line30">line 30</a>
    </li></ul></dd>
    

    

    

    
</dl>


















    
    </div>

    

    

    

    

    

    

    

    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="App.html">App</a></li><li><a href="AuthNav.html">AuthNav</a></li><li><a href="ChannelIcon.html">ChannelIcon</a></li><li><a href="ChannelIconList.html">ChannelIconList</a></li><li><a href="ChannelListPage.html">ChannelListPage</a></li><li><a href="ChannelMemberView.html">ChannelMemberView</a></li><li><a href="ChannelOverview.html">ChannelOverview</a></li><li><a href="ChannelPublicView.html">ChannelPublicView</a></li><li><a href="Feed.html">Feed</a></li><li><a href="MemberList.html">MemberList</a></li><li><a href="NewChannelPage.html">NewChannelPage</a></li><li><a href="NewTaskPage.html">NewTaskPage</a></li><li><a href="PerformancePage.html">PerformancePage</a></li><li><a href="Profile.html">Profile</a></li><li><a href="Proposal.html">Proposal</a></li><li><a href="ProposalsPage.html">ProposalsPage</a></li><li><a href="SignIn.html">SignIn</a></li><li><a href="SignUp.html">SignUp</a></li><li><a href="TaskChecklist.html">TaskChecklist</a></li><li><a href="TaskDetail.html">TaskDetail</a></li><li><a href="TaskEntry.html">TaskEntry</a></li><li><a href="TaskListPage.html">TaskListPage</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc3/jsdoc">JSDoc 3.5.5</a> on Mon Dec 03 2018 00:36:27 GMT-0800 (Pacific Standard Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>