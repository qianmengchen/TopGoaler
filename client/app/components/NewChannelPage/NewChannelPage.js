import React, { Component } from 'react';
import { View, Text } from 'react-native';
import { Input, Button } from 'react-native-elements';
import { board, text, button } from './styles';

class NewChannelPage extends Component {
  _updateName(text) {
    this.setState({ name: text });
  }

  _updateDescription(text) {
    this.setState({ description: text });
  }

  _errorCheck = () => false;

  _submit() {
    this.props.addChannel(this.state.name, this.props.username);
    this.props.navigation.navigate('ChannelListPage');
  }

  constructor(props) {
    super(props);
    this.state = {
      name: '',
      description: ''
    };
  }

  render() {
    return (
      <View style={board.container}>
        <View style={board.title}>
          <Text style={text.title}>New Channel</Text>
        </View>
        <Input
          label="Name"
          labelStyle={text.label}
          errorStyle={{ color: 'red' }}
          errorMessage={
            this._errorCheck(this.state.name)
              ? 'Please enter a valid channel name'
              : null
          }
          containerStyle={board.input}
          value={this.state.name}
          onChangeText={this._updateName.bind(this)}
        />
        <Input
          label="Description"
          labelStyle={text.label}
          errorStyle={{ color: 'red' }}
          errorMessage={
            this._errorCheck(this.state.description)
              ? 'Please enter a valid channel name'
              : null
          }
          containerStyle={board.input}
          value={this.state.description}
          onChangeText={this._updateDescription.bind(this)}
        />
        <View style={board.button}>
          <Button
            title="Create"
            buttonStyle={button.create}
            onPress={this._submit.bind(this)}
          />
        </View>
      </View>
    );
  }
}

export default NewChannelPage;
