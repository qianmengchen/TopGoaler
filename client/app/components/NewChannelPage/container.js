import { connect } from 'react-redux';
import NewChannelPage from './NewChannelPage';
import { createChannelAsUser } from '../../actions';

const mapStateToProps = state => ({
  userId: state.auth.id
});

const mapDispatchToProps = dispatch => ({
  addChannel: (channel, user_id) =>
    dispatch(createChannelAsUser(channel, user_id))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(NewChannelPage);
