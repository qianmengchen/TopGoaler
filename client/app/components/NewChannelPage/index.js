import NewChannelPage from './container';

export { NewChannelPage };
