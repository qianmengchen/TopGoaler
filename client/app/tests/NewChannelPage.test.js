import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunkMiddleware from 'redux-thunk';
import sinon from 'sinon';
import { Button, Input } from 'react-native-elements';
require('isomorphic-fetch');

import NewChannelPage from '../components/NewChannelPage/NewChannelPage';

const middlewares = [thunkMiddleware]; // you can mock any middlewares here if necessary
const mockStore = configureStore(middlewares);

const initialTaskState = {
  tasks: ['do frontend', 'do backend', 'do MVP ftw'],
  isFetching: false,
  currentTaskPage: 'ChannelListPage'
};

describe('Testing New Channel Page', () => {
  const mockNavigation = { navigate: jest.fn() };
  const mockAddChannel = jest.fn();
  const submit = sinon.spy(NewChannelPage.prototype, '_submit');

  const wrapper = shallow(
    <NewChannelPage addChannel={mockAddChannel} navigation={mockNavigation} />,
    {
      context: { store: mockStore(initialTaskState) }
    }
  );
  const render = wrapper.dive();

  it('renders as expected', () => {
    expect(render).toMatchSnapshot();
  });

  it('should have 1 Text field to indicate creating a new channel', () => {
    expect(render.find('Text')).toHaveLength(1);
  });

  it('should have 1 Button to create channel', () => {
    expect(render.find(Button)).toHaveLength(1);
  });

  it('should have 2 Input fields for name and description', () => {
    expect(render.find(Input)).toHaveLength(2);
  });

  it('should invoke corresponding methods when create channel button is pressed', () => {
    const createChannelBtn = render.find(Button);
    createChannelBtn.simulate('press');
    expect(submit.calledOnce).toBe(true);
  });

  it('should correctly update name state when text changes', () => {
    const nameField = wrapper.find(Input).at(0);
    expect(nameField.props().value).toBe('');

    nameField.simulate('changeText', 'notadmin');
    expect(wrapper.state('name')).toBe('notadmin');

    // re-render for updated state
    const nameField2 = wrapper.find(Input).at(0);
    expect(nameField2.props().value).toBe('notadmin');
  });

  it('should correctly update description state when text changes', () => {
    const descriptionField = wrapper.find(Input).at(1);
    expect(descriptionField.props().value).toBe('');

    descriptionField.simulate('changeText', 'notadmin');
    expect(wrapper.state('name')).toBe('notadmin');

    // re-render for updated state
    const descriptionField2 = wrapper.find(Input).at(1);
    expect(descriptionField2.props().value).toBe('notadmin');
  });

  it('should give no error messages when errorCheck fails', () => {
    const nameField = wrapper.find(Input).at(0);
    expect(nameField.props().errorMessage).toBe(null);
  });
});
