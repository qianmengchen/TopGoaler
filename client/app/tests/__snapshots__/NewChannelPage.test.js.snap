// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing New Channel Page renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Component
    style={
      Object {
        "alignItems": "stretch",
        "backgroundColor": "white",
        "flex": 1,
      }
    }
  >
    <Component
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginVertical": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 30,
            "fontWeight": "bold",
          }
        }
      >
        New Channel
      </Text>
    </Component>
    <ForwardRef(Themed.Input)
      containerStyle={
        Object {
          "marginBottom": 10,
          "paddingLeft": 10,
        }
      }
      errorMessage={null}
      errorStyle={
        Object {
          "color": "red",
        }
      }
      label="Name"
      labelStyle={
        Object {
          "color": "grey",
          "fontSize": 15,
          "fontWeight": "300",
        }
      }
      onChangeText={[Function]}
      value=""
    />
    <ForwardRef(Themed.Input)
      containerStyle={
        Object {
          "marginBottom": 10,
          "paddingLeft": 10,
        }
      }
      errorMessage={null}
      errorStyle={
        Object {
          "color": "red",
        }
      }
      label="Description"
      labelStyle={
        Object {
          "color": "grey",
          "fontSize": 15,
          "fontWeight": "300",
        }
      }
      onChangeText={[Function]}
      value=""
    />
    <ForwardRef(Themed.Input)
      containerStyle={
        Object {
          "marginBottom": 10,
          "paddingLeft": 10,
        }
      }
      errorMessage={null}
      errorStyle={
        Object {
          "color": "red",
        }
      }
      label="Image"
      labelStyle={
        Object {
          "color": "grey",
          "fontSize": 15,
          "fontWeight": "300",
        }
      }
      onChangeText={[Function]}
      value="http://shortlink.in/themes/v3/styles/img/url-link.png"
    />
    <Component
      style={
        Object {
          "alignItems": "center",
          "marginVertical": 10,
        }
      }
    >
      <ForwardRef(Themed.Button)
        buttonStyle={
          Object {
            "backgroundColor": "rgba(40, 160, 147, 1)",
            "borderColor": "transparent",
            "borderRadius": 20,
            "borderWidth": 0,
            "height": 40,
            "width": 120,
          }
        }
        onPress={[Function]}
        title="Create"
      />
    </Component>
  </Component>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Component
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "marginVertical": 20,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 30,
                "fontWeight": "bold",
              }
            }
          >
            New Channel
          </Text>
        </Component>,
        <ForwardRef(Themed.Input)
          containerStyle={
            Object {
              "marginBottom": 10,
              "paddingLeft": 10,
            }
          }
          errorMessage={null}
          errorStyle={
            Object {
              "color": "red",
            }
          }
          label="Name"
          labelStyle={
            Object {
              "color": "grey",
              "fontSize": 15,
              "fontWeight": "300",
            }
          }
          onChangeText={[Function]}
          value=""
        />,
        <ForwardRef(Themed.Input)
          containerStyle={
            Object {
              "marginBottom": 10,
              "paddingLeft": 10,
            }
          }
          errorMessage={null}
          errorStyle={
            Object {
              "color": "red",
            }
          }
          label="Description"
          labelStyle={
            Object {
              "color": "grey",
              "fontSize": 15,
              "fontWeight": "300",
            }
          }
          onChangeText={[Function]}
          value=""
        />,
        <ForwardRef(Themed.Input)
          containerStyle={
            Object {
              "marginBottom": 10,
              "paddingLeft": 10,
            }
          }
          errorMessage={null}
          errorStyle={
            Object {
              "color": "red",
            }
          }
          label="Image"
          labelStyle={
            Object {
              "color": "grey",
              "fontSize": 15,
              "fontWeight": "300",
            }
          }
          onChangeText={[Function]}
          value="http://shortlink.in/themes/v3/styles/img/url-link.png"
        />,
        <Component
          style={
            Object {
              "alignItems": "center",
              "marginVertical": 10,
            }
          }
        >
          <ForwardRef(Themed.Button)
            buttonStyle={
              Object {
                "backgroundColor": "rgba(40, 160, 147, 1)",
                "borderColor": "transparent",
                "borderRadius": 20,
                "borderWidth": 0,
                "height": 40,
                "width": 120,
              }
            }
            onPress={[Function]}
            title="Create"
          />
        </Component>,
      ],
      "style": Object {
        "alignItems": "stretch",
        "backgroundColor": "white",
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 30,
                "fontWeight": "bold",
              }
            }
          >
            New Channel
          </Text>,
          "style": Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginVertical": 20,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "New Channel",
            "ellipsizeMode": "tail",
            "style": Object {
              "fontSize": 30,
              "fontWeight": "bold",
            },
          },
          "ref": null,
          "rendered": "New Channel",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "containerStyle": Object {
            "marginBottom": 10,
            "paddingLeft": 10,
          },
          "errorMessage": null,
          "errorStyle": Object {
            "color": "red",
          },
          "label": "Name",
          "labelStyle": Object {
            "color": "grey",
            "fontSize": 15,
            "fontWeight": "300",
          },
          "onChangeText": [Function],
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "containerStyle": Object {
            "marginBottom": 10,
            "paddingLeft": 10,
          },
          "errorMessage": null,
          "errorStyle": Object {
            "color": "red",
          },
          "label": "Description",
          "labelStyle": Object {
            "color": "grey",
            "fontSize": 15,
            "fontWeight": "300",
          },
          "onChangeText": [Function],
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "containerStyle": Object {
            "marginBottom": 10,
            "paddingLeft": 10,
          },
          "errorMessage": null,
          "errorStyle": Object {
            "color": "red",
          },
          "label": "Image",
          "labelStyle": Object {
            "color": "grey",
            "fontSize": 15,
            "fontWeight": "300",
          },
          "onChangeText": [Function],
          "value": "http://shortlink.in/themes/v3/styles/img/url-link.png",
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <ForwardRef(Themed.Button)
            buttonStyle={
              Object {
                "backgroundColor": "rgba(40, 160, 147, 1)",
                "borderColor": "transparent",
                "borderRadius": 20,
                "borderWidth": 0,
                "height": 40,
                "width": 120,
              }
            }
            onPress={[Function]}
            title="Create"
          />,
          "style": Object {
            "alignItems": "center",
            "marginVertical": 10,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "buttonStyle": Object {
              "backgroundColor": "rgba(40, 160, 147, 1)",
              "borderColor": "transparent",
              "borderRadius": 20,
              "borderWidth": 0,
              "height": 40,
              "width": 120,
            },
            "onPress": [Function],
            "title": "Create",
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
        },
        "type": [Function],
      },
    ],
    "type": "View",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Component
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "marginVertical": 20,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 30,
                  "fontWeight": "bold",
                }
              }
            >
              New Channel
            </Text>
          </Component>,
          <ForwardRef(Themed.Input)
            containerStyle={
              Object {
                "marginBottom": 10,
                "paddingLeft": 10,
              }
            }
            errorMessage={null}
            errorStyle={
              Object {
                "color": "red",
              }
            }
            label="Name"
            labelStyle={
              Object {
                "color": "grey",
                "fontSize": 15,
                "fontWeight": "300",
              }
            }
            onChangeText={[Function]}
            value=""
          />,
          <ForwardRef(Themed.Input)
            containerStyle={
              Object {
                "marginBottom": 10,
                "paddingLeft": 10,
              }
            }
            errorMessage={null}
            errorStyle={
              Object {
                "color": "red",
              }
            }
            label="Description"
            labelStyle={
              Object {
                "color": "grey",
                "fontSize": 15,
                "fontWeight": "300",
              }
            }
            onChangeText={[Function]}
            value=""
          />,
          <ForwardRef(Themed.Input)
            containerStyle={
              Object {
                "marginBottom": 10,
                "paddingLeft": 10,
              }
            }
            errorMessage={null}
            errorStyle={
              Object {
                "color": "red",
              }
            }
            label="Image"
            labelStyle={
              Object {
                "color": "grey",
                "fontSize": 15,
                "fontWeight": "300",
              }
            }
            onChangeText={[Function]}
            value="http://shortlink.in/themes/v3/styles/img/url-link.png"
          />,
          <Component
            style={
              Object {
                "alignItems": "center",
                "marginVertical": 10,
              }
            }
          >
            <ForwardRef(Themed.Button)
              buttonStyle={
                Object {
                  "backgroundColor": "rgba(40, 160, 147, 1)",
                  "borderColor": "transparent",
                  "borderRadius": 20,
                  "borderWidth": 0,
                  "height": 40,
                  "width": 120,
                }
              }
              onPress={[Function]}
              title="Create"
            />
          </Component>,
        ],
        "style": Object {
          "alignItems": "stretch",
          "backgroundColor": "white",
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 30,
                  "fontWeight": "bold",
                }
              }
            >
              New Channel
            </Text>,
            "style": Object {
              "alignItems": "center",
              "justifyContent": "center",
              "marginVertical": 20,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "New Channel",
              "ellipsizeMode": "tail",
              "style": Object {
                "fontSize": 30,
                "fontWeight": "bold",
              },
            },
            "ref": null,
            "rendered": "New Channel",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "containerStyle": Object {
              "marginBottom": 10,
              "paddingLeft": 10,
            },
            "errorMessage": null,
            "errorStyle": Object {
              "color": "red",
            },
            "label": "Name",
            "labelStyle": Object {
              "color": "grey",
              "fontSize": 15,
              "fontWeight": "300",
            },
            "onChangeText": [Function],
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "containerStyle": Object {
              "marginBottom": 10,
              "paddingLeft": 10,
            },
            "errorMessage": null,
            "errorStyle": Object {
              "color": "red",
            },
            "label": "Description",
            "labelStyle": Object {
              "color": "grey",
              "fontSize": 15,
              "fontWeight": "300",
            },
            "onChangeText": [Function],
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "containerStyle": Object {
              "marginBottom": 10,
              "paddingLeft": 10,
            },
            "errorMessage": null,
            "errorStyle": Object {
              "color": "red",
            },
            "label": "Image",
            "labelStyle": Object {
              "color": "grey",
              "fontSize": 15,
              "fontWeight": "300",
            },
            "onChangeText": [Function],
            "value": "http://shortlink.in/themes/v3/styles/img/url-link.png",
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <ForwardRef(Themed.Button)
              buttonStyle={
                Object {
                  "backgroundColor": "rgba(40, 160, 147, 1)",
                  "borderColor": "transparent",
                  "borderRadius": 20,
                  "borderWidth": 0,
                  "height": 40,
                  "width": 120,
                }
              }
              onPress={[Function]}
              title="Create"
            />,
            "style": Object {
              "alignItems": "center",
              "marginVertical": 10,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "buttonStyle": Object {
                "backgroundColor": "rgba(40, 160, 147, 1)",
                "borderColor": "transparent",
                "borderRadius": 20,
                "borderWidth": 0,
                "height": 40,
                "width": 120,
              },
              "onPress": [Function],
              "title": "Create",
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
          },
          "type": [Function],
        },
      ],
      "type": "View",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
